port/unix/omrsysinfo.c

```diff
@@ -1488,10 +1488,27 @@ retrieveLinuxCgroupMemoryStats(struct OMRPortLibrary *portLibrary, struct OMRCgr
 	}
 	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, CGROUP_MEMORY_SWAP_LIMIT_IN_BYTES_FILE, numItemsToRead, "%lu", &cgroupMemInfo->memoryAndSwapLimit);
 	if (0 != rc) {
-		goto _exit;
+		if (OMRPORT_ERROR_FILE_NOENT == rc) {
+			/* It is possible file memory.memsw.limit_in_bytes is not present if
+			 * swap space is not configured. In such cases, set memoryAndSwapLimit to same as memoryLimit.
+			 */
+			cgroupMemInfo->memoryAndSwapLimit = cgroupMemInfo->memoryLimit;
+			rc = 0;
+		} else {
+			goto _exit;
+		}
 	}
 	rc = readCgroupSubsystemFile(portLibrary, OMR_CGROUP_SUBSYSTEM_MEMORY, CGROUP_MEMORY_SWAP_USAGE_IN_BYTES_FILE, numItemsToRead, "%lu", &cgroupMemInfo->memoryAndSwapUsage);
 	if (0 != rc) {
+		if (OMRPORT_ERROR_FILE_NOENT == rc) {
+			/* It is possible file memory.memsw.usage_in_bytes is not present if
+			 * swap space is not configured. In such cases, set memoryAndSwapUsage to memoryUsage.
+			 */
+			cgroupMemInfo->memoryAndSwapUsage = cgroupMemInfo->memoryUsage;
```

Comment:
what is cgroupMemInfo->memoryAndSwapUsage supposed to represent?

Reply:
It represents sum of memory and swap usage. From `port/unix_include/omrcgroup.h`:
```
        uint64_t memoryAndSwapUsage; /**< current memory + swap usage in bytes (as in memory.memsw.usage_in_bytes file) */
```